frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -1163,4 +1154,88 @@ private String getProjectDefaultLicenseInfoHeaderText() {
             return "";
         }
     }
+
+    private void serveProjectList(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
+        Map<String, Integer> projectParameter = getServeListMapFromRequest(request);
+        List<Project> projectList = getFilteredProjectList(request);
+        JSONArray jsonProjects = getProjectData(projectList, projectParameter);
+
+        JSONObject jsonResult = createJSONObject();
+        jsonResult.put(DATATABLE_RECORDS_TOTAL, projectList.size());
+        jsonResult.put(DATATABLE_RECORDS_FILTERED, projectList.size());
+        jsonResult.put(DATATABLE_DISPLAY_DATA, jsonProjects);
+
+        try {
+            writeJSON(request, response, jsonResult);
+        } catch (IOException e) {
+            log.error("Problem rendering RequestStatus", e);
+        }
+    }
+
+    public JSONArray getProjectData(List<Project> projectList, Map<String, Integer> projectParameter) {
+        List<Project> sortedProjects = sortProjectList(projectList, projectParameter);
+        int startIdx = projectParameter.get(DATATABLE_DISPLAY_START);
+        int listDisplayAmount = projectParameter.get(DATATABLE_DISPLAY_LENGTH);
+        int count = listDisplayAmount == -1 ? sortedProjects.size() : startIdx + listDisplayAmount;
+
+        JSONArray projectData = createJSONArray();
+        for (int i = startIdx; i < count; i++) {
+            if (i < sortedProjects.size()) {
+                JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+                Project project = sortedProjects.get(i);
+                jsonObject.put("id", project.getId());
+                jsonObject.put("DT_RowId", project.getId());
+                jsonObject.put("name", SW360Utils.printName(project));
+                jsonObject.put("desc", project.isSetDescription() ? project.getDescription() : "");
+                jsonObject.put("state", project.isSetState() ? project.getState().toString() : "");
+                jsonObject.put("cState", project.isSetClearingState() ? project.getClearingState().toString() : "");
+                jsonObject.put("clearing", "Not loaded yet");
+                jsonObject.put("resp", project.isSetProjectResponsible() ? project.getProjectResponsible() : "");
+                jsonObject.put("lProjSize", String.valueOf(project.getLinkedProjectsSize()));
+                jsonObject.put("lRelsSize", String.valueOf(project.getReleaseIdToUsageSize()));
+                jsonObject.put("attsSize", String.valueOf(project.getAttachmentsSize()));
+                projectData.put(jsonObject);
+            }
+        }
+
+        return projectData;
+    }
+
+    private List<Project> sortProjectList(List<Project> projectList, Map<String, Integer> projectParameter) {
+        List<Project> sortedList;
+
+        switch (projectParameter.get(DATATABLE_SORT_COLUMN)) {
+            case 0: // Project Name
```

Comment:
It is easier to read, if you extract the number to a field in some class and just reference `PROJECT_NAME_ROW` here

Reply:
changed to constant values.